### PR TITLE
Activity feed prototype

### PR DESCRIPTION
**Pull request by ksingarajah**
Implemented the activity feed prototype. Added the URL mapping, servlet, and JSP file. Right now it just displays "this is the activity feed".

### CODE REVIEW

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  // Forwards the request to activity-feed.jsp.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
```
Comment (tsalako) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: Can we get the user here?

Comment (tsalako) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: So we can verify that we can use the userStore properly

**Review by tsalako — CHANGES_REQUESTED** [review]
Looks good! Very easy code to read.

Two things:
1. Can we set up the testing for this new page?
2. Can you include a screenshot of this page as a comment to this pull request?

File: src/main/webapp/WEB-INF/view/activityfeed.jsp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <title>Activity Feed</title>
+    <link rel="stylesheet" href="/css/main.css">
+</head>
+
+<body>
+
+    <nav>
+        <a id="navTitle" href="/">CodeU Chat App</a>
+        <a href="/conversations">Conversations</a>
+        <% if(request.getSession().getAttribute("user") != null) { %>
```
Comment (ricglz) [inline src/main/webapp/WEB-INF/view/activityfeed.jsp:14]: The indentation is a little bit weird, but maybe is the Github editor.

**Comment by ricglz** [comment]
It is really cool. Easy to read code, no unnecessary code.

**Review by ricglz — APPROVED** [review]

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 1 above)
Comment (ksingarajah) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: As in add a getUserStore() method?

File: src/main/webapp/WEB-INF/view/activityfeed.jsp (hunk 2 above)
Comment (ksingarajah) [reply src/main/webapp/WEB-INF/view/activityfeed.jsp:14]: I think it was the auto-indentation of my text editor, will fix.

**Comment by ksingarajah** [comment]
screen shot 2018-05-29 at 1 05 46 pm

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 1 above)
Comment (tsalako) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: UserStore#getUser actually. So getting the user from the userstore itself. Similar to how it is done in ChatServlet#doPost, and ConversationServlet#doPost.

**Review by tsalako — CHANGES_REQUESTED** [review]
Hey. How are users going to navigate to this page? We need to add this page to the navbar. To do this we only have to update the jsp files to have an option for this page

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 1 above)
Comment (ksingarajah) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: Resolved.

File: src/main/webapp/WEB-INF/view/activityfeed.jsp (hunk 2 above)
Comment (ksingarajah) [reply src/main/webapp/WEB-INF/view/activityfeed.jsp]: Resolved.

Comment (ksingarajah) [reply src/main/webapp/WEB-INF/view/activityfeed.jsp:14]: Resolved.

File: src/test/java/codeu/controller/ActivityFeedServletTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        mockResponse = Mockito.mock(HttpServletResponse.class);
+        mockUserStore = Mockito.mock(UserStore.class);
+
+        Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp"))
+        .thenReturn(mockRequestDispatcher);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+        activityFeedServlet.setUserStore(mockUserStore);
+    }
+
+    @ Test
```
Comment (tsalako) [inline src/test/java/codeu/controller/ActivityFeedServletTest.java]: There shouldn't be a space between @ and Test. Also there shouldn't be an empty line between the @Test annotation and the method. (Ditto to all the tests below.)

Comment (ksingarajah) [reply src/test/java/codeu/controller/ActivityFeedServletTest.java]: Fixed!

**Comment by tsalako** [comment]
Alright. Im going to approve but just make sure the test pass when you run "mvn test". Also, I realize that this page isn't going to show up in the navbar unless you are on that page already currently. We can discuss in the meeting today how we are going to follow this up with the navbar change.

**Review by tsalako — APPROVED** [review]

**Pull request merged**